### PR TITLE
let's document the code quality tools we have set up

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: coccinelle/run-coccinelle.sh [hunk 1]
```diff
@@ -10,7 +11,7 @@ case "$1" in
                 ;;
 esac
 
-for SCRIPT in ${@-coccinelle/*.cocci} ; do
+for SCRIPT in $top/coccinelle/*.cocci ; do
```
Comment (keszybz) [inline coccinelle/run-coccinelle.sh]: This removes the ability to run a single script or some subset of them. I use this :(

File: doc/CODE_QUALITY.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Code Quality Tools
+
+The systemd project has a number of code quality tools set up in the source
+tree and on the github infrastructure. Here's an incomprehensive list of the
+available functionality.
+
+1. Use `ninja -C build test` to run the unit tests. Some tests are skipped if
+   no privileges are available, hence consider also running them with `sudo
+   ninja -C build test`.
```
Comment (keszybz) [inline doc/CODE_QUALITY.md]: Maybe also mention `-Dslow-tests=true` and `-Dtests=unsafe` and manual tests?

File: doc/CODE_QUALITY.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+   `./tools/find-tabs.sh recpatch` to fix them.
+
+6. Use `ninja -C build check-api-docs` to compare the the list of exported
+   symbols of `libsystemd.so` and `libudev.so` with the list of man pages. Symbols
+   lacking documentation are highlighted.
+
+7. Use `ninja -C build hwdb-update` to automatically download and import the
+   PCI, USB and OUI databases into hwdb.
+
+8. Use `ninja -C build man/update-man-rules` to update the meson rules for
```
Comment (keszybz) [inline doc/CODE_QUALITY.md:39]: A somewhat unrelated thing, but I can't get used to this name. Is it just me? Maybe we can rename the target to something like 'update-man-list' ?

File: doc/CODE_QUALITY.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+   building man pages automatically from the docbook XML files included in
+   `man/`.
+
+9. There are multiple CI systems in use that run on every github PR submission.
+
+10. Coverity is analyzing systemd master in regular intervals. The reports are
+    available [online](https://scan.coverity.com/projects/systemd).
```
Comment (keszybz) [inline doc/CODE_QUALITY.md]:
```markdown
11. oss-fuzz is continuously fuzzing the codebase. Reports are available [online](https://oss-fuzz.com/v2/testcases?project=systemd).

Access to coverity and oss-fuzz reports is limited. Please reach out the the maintainers is you need access.
```

File: coccinelle/run-coccinelle.sh (hunk 1 above)
Comment (poettering) [reply coccinelle/run-coccinelle.sh]: hmm, will fix

File: doc/CODE_QUALITY.md (hunk 2 above)
Comment (poettering) [reply doc/CODE_QUALITY.md]: do we have a list of manual tests?

File: doc/CODE_QUALITY.md (hunk 3 above)
Comment (poettering) [reply doc/CODE_QUALITY.md:39]: sure, the "man/" prefix is really weird...

**Comment by poettering** [comment]
Pushed a new version, with all suggested changes included. Also adding one minor follow-up commit. PTAL.

**Pull request merged**